lib/util/selector-parser.js

```diff
@@ -0,0 +1,255 @@
+"use strict";
+
+//   selectors = selector [ws ',' ws selector]*;
```

Comment:
I didn't add support for `:exit` as that would require parser to understand a construct like that. Not 100% sure it's needed, but if it is, it can be added at later iterations, I think.